### PR TITLE
j.core.state tests

### PR DESCRIPTION
**Pull request by ahmedelsayed-93**
#### What this PR resolves:
implement tests to cover j.core.state methods

#### Changes proposed in this PR:

-
-

**Version**: 

**Fixes**: #

### CODE REVIEW

**Comment by ahmedelsayed-93** [comment]
```
root@js9all:~/core9/tests/jumpscale_test# nosetests -s -v test02_j_state.py 
JS-004 ... [Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
ok
JS-005 ... [Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
ok
JS-005 ... SKIP: https://github.com/Jumpscale/core9/issues/157
JS-006 ... [Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
ok
JS-007 ... [Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/state.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/state.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/state.toml
ok
JS-008 ... [Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/state.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/state.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/state.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/state.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/state.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/state.toml
ok
JS-008 ... [Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/jumpscale9.toml
[Tue09 09:53] - ExecutorBase.py     :433 :j.executor.localhost           - DEBUG    - file write:/root/js9host/cfg/state.toml
ok
JS-010 ... SKIP: https://github.com/Jumpscale/core9/issues/159

----------------------------------------------------------------------
Ran 8 tests in 0.484s

OK (SKIP=2)
```

File: tests/jumpscale_test/test02_j_state.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+        self.lg.info('Get the value of an non-existing key from existing dict with default value')
+        value = self.client.configGetFromDictBool('new_dict', 'new_dict_key', default=1)
+        self.assertEqual(value, 'new_dict_value')
+        self.assertTrue(isinstance(value, bool))
+
+    def test004_config_set(self):
+        """ JS-006
+        **Test Scenario:**
+        #. Set new key, value with save equalt to true.
```
Comment (islamTaha12) [inline tests/jumpscale_test/test02_j_state.py]: equal not equalt

File: tests/jumpscale_test/test02_j_state.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
+from js9 import j
```
[the remark this answers is not in the document]
Comment (ahmedelsayed-93) [reply tests/jumpscale_test/test02_j_state.py]: we need to import it if we want to run form nosetest

File: tests/jumpscale_test/test02_j_state.py (hunk 1 above)
Comment (ahmedelsayed-93) [reply tests/jumpscale_test/test02_j_state.py]: done

File: tests/jumpscale_test/test02_j_state.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.client = j.core.state
+        cls.config_path = '/root/js9host/cfg/jumpscale9.toml'
+        cls.config_file_content = cls.get_config()
+
+    @classmethod
+    def tearDownClass(cls):
```
[the remark this answers is not in the document]
Comment (ahmedelsayed-93) [reply tests/jumpscale_test/test02_j_state.py]: done

File: tests/jumpscale_test/test02_j_state.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (islamTaha12) [reply tests/jumpscale_test/test02_j_state.py]: that is right.

**Pull request merged**